### PR TITLE
[PURR-149] - string conversion not handled properly

### PR DESCRIPTION
**Pull request by dbenham**
https://purr.purdue.edu/support/ticket/2482 - original ticket

Looks like an untested quote fix from an earlier change. A php string conversion issue. Reworked the SQL statement being built not to use embedded string variables inside a double quoted string. Made the code match what was in the rest of the file.

Got a pretty good write up in the Jira card on reproducing. Easily able to reproduce error, make fix, and observe fix. 

There were two other fixes made in dev that were added to this as well that were not part of the original ticket, they followed the convention in the file, so changes were not needed for these.

No hotfix needed, part of the 2.2.27 build

### CODE REVIEW

File: core/components/com_publications/helpers/tags.php [hunk 1]
```diff
@@ -631,9 +635,10 @@ public function getTagUsage($tag, $rtrn='id')
 	 */
 	public function countTags($id)
 	{
-		$sql = "SELECT COUNT(*) FROM $this->_tag_tbl AS t,
-			$this->_obj_tbl AS rt WHERE rt.objectid=$this->_db->quote($id)
-			AND rt.tbl='$this->_tbl' AND rt.tagid=t.id";
+		$sql  = "SELECT COUNT(*) FROM $this->_tag_tbl AS t, ";
+		$sql .= $this->_obj_tbl . " AS rt WHERE rt.objectid=" . $this->_db->quote($id);
+		$sql .= "AND rt.tbl='$this->_tbl' AND rt.tagid=t.id";
```
Comment (nkissebe) [inline core/components/com_publications/helpers/tags.php:640]: shouldn't we use quote() here?

File: core/components/com_publications/helpers/tags.php [hunk 2]
```diff
@@ -677,7 +682,7 @@ public function getPickedTags($tags = array())
 	 */
 	public function getFOSTag($tagid)
 	{
-		$sql = "SELECT jto.tagid FROM $this->_tag_tbl AS jt LEFT JOIN $this->_obj_tbl AS jto ON jto.objectid = jt.id WHERE jto.tbl =" . '"tags" AND jto.label = "parent" AND jt.id=' . $this->_db->quote($tagid);
+		$sql = "SELECT jto.tagid FROM $this->_tag_tbl AS jt LEFT JOIN $this->_obj_tbl AS jto ON jto.objectid = jt.id WHERE jto.tbl =" . '"tags" AND jto.label = "parent" AND jt.id=' . $tagid;
```
Comment (nkissebe) [inline core/components/com_publications/helpers/tags.php:685]: Why did we remove the use of quote()

File: core/components/com_publications/helpers/tags.php [hunk 3]
```diff
@@ -686,7 +691,7 @@ public function getFOSTag($tagid)
 			$fosTagObjects = [];
 			foreach ($ids as $id)
 			{
-				$sql = "SELECT jt.* FROM $this->_tag_tbl AS jt WHERE jt.id = " . $this->_db->quote($id);
+				$sql = "SELECT jt.* FROM $this->_tag_tbl AS jt WHERE jt.id = $id";
```
Comment (nkissebe) [inline core/components/com_publications/helpers/tags.php:694]: why remove the use of quote()

File: core/components/com_publications/helpers/tags.php [hunk 4]
```diff
@@ -753,7 +758,7 @@ public function getSuggestedTags($pubtitle = '', $typealias = '', $attached_tags
 			$w = 1;
 			foreach ($keywords as $key)
 			{
-				$sql .= 't.raw_tag LIKE "%'.$this->_db->escape($key).'%"';
+				$sql .= 't.raw_tag LIKE "%'.$this->_db->quote($key).'%"';
```
Comment (nkissebe) [inline core/components/com_publications/helpers/tags.php:761]: I don't think you can use quote() here, seems like we changed it to escape() recently

Comment (rw2) [reply core/components/com_publications/helpers/tags.php:761]: "can't" or "shouldn't"

Comment (nkissebe) [reply core/components/com_publications/helpers/tags.php:761]: %'name'%  won't give the results you want in a LIKE

File: core/components/com_publications/helpers/tags.php (hunk 1 above)
Comment (dbenham) [reply core/components/com_publications/helpers/tags.php:640]: You mean around $this->_tbl ?

File: core/components/com_publications/helpers/tags.php (hunk 2 above)
Comment (dbenham) [reply core/components/com_publications/helpers/tags.php:685]: b/c we're an idiot

File: core/components/com_publications/helpers/tags.php (hunk 1 above)
Comment (dbenham) [reply core/components/com_publications/helpers/tags.php:640]: @nkissebe Please comment on this so I can close this PR out. Thanks.

Comment (nkissebe) [reply core/components/com_publications/helpers/tags.php:640]: You might need a space before the "AND" ...

Comment (dbenham) [reply core/components/com_publications/helpers/tags.php:640]: nitpicker ;)

**Comment by dbenham** [comment]
@nkissebe - Need you to approve, I can't assign myself as a reviewer, and I do not want to uncheck "require approvals" for the dev branch. And as far as I can tell, github does not allow you to add yourself as a reviewer of your own pull request.

**Review by nkissebe — APPROVED** [review]

**Comment by dbenham** [comment]
closing this PR, I totally messed up the merge with another commit, I'm just going to start over.

**Pull request closed without merging**